### PR TITLE
Fixed unit tests to run with venus and jasmine-node. Added README.md for tests directory

### PR DESCRIPTION
**Pull request by kate2753**
All of the unit tests now can be ran with jasmine in node and with venus. All unit tests pass.

Added README file with directions on how to run unit tests

### CODE REVIEW

File: test/specs/dust-core-plus-filters.spec.js [hunk 1]
```diff
@@ -1,17 +1,27 @@
...
 describe("dust core plus filters works", function() {
   it("should have dust and dust filters defined", function(){
     expect(dust).not.toBeUndefined();
-    expect(dustFilters).toEqual(jasmine.any(Object));
+    expect(dust.filters).toEqual(jasmine.any(Object));
   });
-  it("should have my custom dust filters defined", function(){
-    /* TODO: replace this with something good */
-    expect(typeof dustFilters.temp).toEqual("function");
+  it("should have all filers that original filters had", function(){
```
Comment (jimmyhchan) [inline test/specs/dust-core-plus-filters.spec.js]: typo. /filers/filters/

File: test/specs/filter-JsonStringify.spec.js [hunk 2]
```diff
@@ -1,11 +1,24 @@
...
+ * @venus-include ../setup/old-filters.js
+ * @venus-include ../../lib/dust-filters-secure.js
+ * @venus-include ../setup/simple-json.js
+ * @venus-fixture xss.fixture.html
+ */
+
+var arrayOfJsStrings,
+    customJSONStringify = dust.filters.js;
+
+if (typeof module !== 'undefined' && module.exports) {
```
Comment (jimmyhchan) [inline test/specs/filter-JsonStringify.spec.js:14]: (editted)

seems we can use this and drop the more verbose ...
typeof module !== 'undefined' && typeof module.exports !== 'undefined'

File: test/specs/filter-escapeHTML.spec.js [hunk 3]
```diff
@@ -12,20 +12,29 @@
...
 }
 
-var testStrings = arrayOfSimpleStrings.concat(arrayOfBadStrings).concat(arrayOfSimpleHTMLStrings).concat(arrayOfBadHTMLStrings),
-    dustFilters = dust.filters;
+  testStrings = arrayOfSimpleStrings
+                   .concat(arrayOfBadStrings)
+                   .concat(arrayOfSimpleHTMLStrings)
+                   .concat(arrayOfBadHTMLStrings);
+
+if (typeof module == 'undefined'){
```
Comment (jimmyhchan) [inline test/specs/filter-escapeHTML.spec.js:35]: typeof module !== 'undefined' && module.exports

File: test/specs/filter-escapeUrl.spec.js [hunk 4]
```diff
@@ -1,15 +1,37 @@
...
+ * @venus-include ../setup/bad-urls.js
+ * @venus-fixture xss.fixture.html
+ */
 
+var arrayOfSimpleStrings,
+    arrayOfBadStrings,
+    arrayOfSimpleUrls,
+    arrayOfBadUrls;
+
+if (typeof module !== 'undefined' && module.exports) {
```
Comment (jimmyhchan) [inline test/specs/filter-escapeUrl.spec.js:19]: typeof module !== 'undefined' && typeof module.exports !== 'undefined'

**Comment by jimmyhchan** [comment]
This is awesome!

There's actually a Makefile in the root directory that we should update as well. 

Thanks

File: Makefile [hunk 5]
```diff
@@ -1,15 +1,26 @@
...
-# Run jasmine-test
+# Run unit tests
 #
-test-venus:
-	node node_modules/venus/bin/venus run -t test/specs/filter-escapeHTML.spec.js
 
-test-jasmine:
-# node test/server/specRunner.js  --color test/specs/ 
-	node node_modules/jasmine-node/bin/jasmine-node test/specs/filter-escapeHTML.spec.js
+* install nodejs 0.6 or greater
```
Comment (jimmyhchan) [inline Makefile]: Comments in make files should use #

File: Makefile [hunk 6]
```diff
@@ -1,15 +1,26 @@
 #
-# Run jasmine-test
+# Run unit tests
 #
-test-venus:
```
Comment (jimmyhchan) [inline Makefile:4]: we use to be able to run
`make test-venus`
from the project root. It'll be nice to have that again.

File: Makefile [hunk 7]
```diff
@@ -1,15 +1,26 @@
...
-	node node_modules/venus/bin/venus run -t test/specs/filter-escapeHTML.spec.js
 
-test-jasmine:
```
Comment (jimmyhchan) [inline Makefile:7]: ditto

File: lib/dust-filters-secure.js [hunk 8]
```diff
@@ -98,7 +98,7 @@
           replace(/&lt;/g, '<').
           replace(/&gt;/g, '>').
           replace(/&quot;/g, '"').
-          replace(/&#xfffd;/g, '\u007f').
```
Comment (jimmyhchan) [inline lib/dust-filters-secure.js:101]: This was a decision on my part ... and this badly needs documentation 

Here's what i implemented: 
Line 51: the delete character u007f is changed to a really ugly xfffd character
Line 57: control characters 0x00 - 0x20 will be changed to a really ugly xfffd character

when we unencode, we don't know what to unencode to. 
IMO, it's better to unencode to the less ugly 007f character.

xfffd is used 
http://www.fileformat.info/info/unicode/char/fffd/index.htm

Comment (kate2753) [inline lib/dust-filters-secure.js:101]: Makes sense, I'll revert my change. Thanks!

File: Makefile (hunk 5 above)
Comment (kate2753) [inline Makefile]: Fixed. Thanks!

File: Makefile (hunk 6 above)
Comment (kate2753) [inline Makefile:4]: Fixed. Thanks!

File: Makefile (hunk 7 above)
Comment (kate2753) [inline Makefile:7]: Fixed. Thanks!

**Pull request merged**